### ISSUE TITLE
The request body must contain the following parameter: 'scope'.

### ISSUE THREAD

**Issue by devtoro** [Innoactive/react-oauth2-auth-code-flow#1]
Hi,

I am trying to implement OAuth2 code grant flow with Microsoft as my authoriser (Azure AD) but there seems to be some kind of issue with the add-on.

I am intializing the  ClientOAuth2 as suggested in the wiki:

```
const oauthClient = new ClientOAuth2({
    clientId: 'XXXXXXXXXX',
    authorizationUri: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
    redirectUri: "http://localhost:3001/spa_oauth_redirect",
    accessTokenUri: '',
    clientSecret: '',
    scopes: [ /list of scopes/],
  });
```

and I get the following response:

**The request body must contain the following parameter: 'scope'.**

I tried adding the **args** prop to the **RequestAuthorizationCode** component with the object `{ scope: [/list of scopes/] }` as value, but again I got the same error.

Any ideas why this might be happening?

Thanks!

**Comment by burnedikt** [comment]
That's weird indeed. It seems like the [underlying `getToken`](https://github.com/mulesoft-labs/js-client-oauth2/blob/68d2a71565641d321148798d7df9d78a3182e584/src/client-oauth2.js#L593-L654) method of [client-oauth2](https://github.com/mulesoft-labs/js-client-oauth2) ignores the defined scopes and does not pass them on to the server. 

We could mitigate this here, by explicitly defining the scopes as part of the body in this line ... https://github.com/Innoactive/react-oauth2-auth-code-flow/blob/f5b6995287075e2c4d678a57c2050d6f22a8957e/src/index.js#L94

... but I wonder whether there's a reason why scopes are not sent along per default in the token request of the authcode flow (but in all other flows).

**Comment by burnedikt** [comment]
Indeed, [RFC 6479](https://tools.ietf.org/html/rfc6749#section-4.1.3) does not state that `scope` is to be sent to the authorization server with the access token request but only in the authorization request (where it would never be sent as part of the body but always as [part of the url](https://tools.ietf.org/html/rfc6749#section-4.1.1), though).

From your description, I can't really judge whether the issue happens during the authorization request or the access token request. 

---

If it happens during the authorization request, passing the scopes to the `RequestAuthorizationCode` component like 

```jsx
<RequestAuthorizationCode args={{scope: [/list of scopes/]}} />
```

should work (just like you tried) - although you are right in a way that this library here should automatically add the scopes if defined on the `ClientOAuth2` instance

---

If the error comes as a response to the access token request, this would be some unexpected behaviour of Azure requiring the `scope` parameter but it could be mitigated as[ described above](https://github.com/Innoactive/react-oauth2-auth-code-flow/issues/1#issuecomment-777547953).
